include/nccl_ofi_rdma.h

```diff
@@ -758,6 +767,35 @@ struct nccl_net_ofi_rdma_ep {
 	/* thread id of the thread that called get_ep().  Used as the
 	   hash key for the endpoint hash */
 	long creating_thread_id;
+
+
+	/** 
+	 * Overriding base virtual functions
+	 */
+
+	int listen(nccl_net_ofi_conn_handle_t *handle,
+		   nccl_net_ofi_listen_comm_t **listen_comm) override;
+
+
+	/**
+	 * @brief	Execute the connect functionality from listen/connect/accept
+	 *		connection establishment
+	 *
+	 * The connect functionality does the following: (a) create send communicator
+	 * with only the first communicator rail being initalized, (b) post send
+	 * operation to send connect message to remote, containing local endpoint
+	 * addresses, (c) wait until message is delivered, (d) waits for the connect
+	 * response message, and (e) calls finish_connect.
+	 *
+	 * The `finish_connect' method completes the initialization of the remaining
+	 * communicator rails using the received connect responce message.
+	 */
+	int connect(nccl_net_ofi_conn_handle_t *handle,
+		    nccl_net_ofi_send_comm_t **send_comm) override;
+
+
+	int release_ep(bool skip_lock, bool force_cleanup) override;
```

Comment:
Please add a todo to your list to make this just a destructor.  I know what it's not now, but we need to fix the locking before we call this done.

Reply:
I currently see `release_ep` as a wrapper for the destructor that updates the `ref_cnt` and checks whether to delete the endpoint or not, and have trouble seeing how it should be refactored into destructor (since it only conditionally deletes). Are your thoughts that I move grabbing the domain lock into the destructor and move the `ref_cnt` update and check up to whatever function is currently calling `release_ep`, or something else?

Reply:
yeah, that's true today, but it's one of those things that should go away as we refactor.  So not necessary for this PR, but something we should do before we're done.